src/main/java/ch/njol/skript/structures/StructFunction.java

```diff
@@ -0,0 +1,64 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.structures;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.function.Function;
+import ch.njol.skript.lang.function.FunctionEvent;
+import ch.njol.skript.lang.function.Functions;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+public class StructFunction extends PreloadingStructure {
```

Comment:
Wouldn't this eliminate the use of options inside functions, because StructOptions is a normal structure and StrucFunction a PreloadingStructure? Again, a priority system is probably better.

Reply:
It wouldn't eliminate them, because options currently don't work in function signatures: https://github.com/SkriptLang/Skript/issues/2492
I'll make StructOptions a PreloadingStructure too, and make it apply the options during preload, and during normal load.

Reply:
I think a priority system in general would be better and less hacky.